test/infrastructure/docker/api/v1beta1/devmachine_types.go

```diff
@@ -66,6 +66,122 @@ const (
+	// for the control plane to be initialized.
+	DevMachineInMemoryVMWaitingForControlPlaneInitializedV1Beta2Reason = clusterv1.WaitingForControlPlaneInitializedV1Beta2Reason
+
+	// DevMachineInMemoryVMWaitingForBootstrapDataV1Beta2Reason documents a fake VM for a DevMachine's in memory backend waiting for the bootstrap
+	// data to be ready.
+	DevMachineInMemoryVMWaitingForBootstrapDataV1Beta2Reason = clusterv1.WaitingForBootstrapDataV1Beta2Reason
+
+	// DevMachineInMemoryVMWaitingForStartupTimeoutV1Beta2Reason documents when the infrastructure for a fake VM for a DevMachine's in memory backend
+	// is provisioning (it waits for a startup timeout).
+	DevMachineInMemoryVMWaitingForStartupTimeoutV1Beta2Reason = "WaitingForStartupTimeout"
+
+	// DevMachineInMemoryVMProvisionedV1Beta2Reason documents when a fake VM for a DevMachine's in memory backend is fully provisioned.
+	DevMachineInMemoryVMProvisionedV1Beta2Reason = clusterv1.ProvisionedV1Beta2Reason
+
+	// DevMachineInMemoryVMInternalErrorV1Beta2Reason surfaces unexpected error when reconciling a fake VM for a DevMachine's in memory backend.
+	DevMachineInMemoryVMInternalErrorV1Beta2Reason = clusterv1.InternalErrorV1Beta2Reason
+)
+
+// NodeProvisioned condition and corresponding reasons that will be used in v1Beta2 API version for a DevMachine's in memory backend.
+const (
+	// DevMachineInMemoryNodeProvisionedV1Beta2Condition documents the status of the provisioning of the node hosted on the DevMachine's in memory backend.
+	DevMachineInMemoryNodeProvisionedV1Beta2Condition string = "NodeProvisioned"
+
+	// DevMachineInMemoryNodeWaitingForVMProvisionedV1Beta2Reason documents a fake Node for a DevMachine's in memory backend waiting for the VM
+	// to be provisioned.
+	DevMachineInMemoryNodeWaitingForVMProvisionedV1Beta2Reason = "WaitingForVMProvisioned"
+
+	// DevMachineInMemoryNodeWaitingForStartupTimeoutV1Beta2Reason documents when the fake Node for a DevMachine's in memory backend
+	// is provisioning (it waits for a startup timeout).
+	DevMachineInMemoryNodeWaitingForStartupTimeoutV1Beta2Reason = "WaitingForStartupTimeout"
+
+	// DevMachineInMemoryNodeProvisionedV1Beta2Reason documents when a fake Node for a DevMachine's in memory backend is fully provisioned.
+	DevMachineInMemoryNodeProvisionedV1Beta2Reason = clusterv1.ProvisionedV1Beta2Reason
+
+	// DevMachineInMemoryNodeInternalErrorV1Beta2Reason surfaces unexpected error when reconciling a fake Node for a DevMachine's in memory backend.
+	DevMachineInMemoryNodeInternalErrorV1Beta2Reason = clusterv1.InternalErrorV1Beta2Reason
+)
+
+// EtcdProvisioned condition and corresponding reasons that will be used in v1Beta2 API version for a DevMachine's in memory backend.
+const (
+	// DevMachineInMemoryEtcdProvisionedV1Beta2Condition documents the status of the provisioning VM implementing the DevMachine's in memory backend.
```

Comment:
```suggestion
	// DevMachineInMemoryEtcdProvisionedV1Beta2Condition documents the status of the provisioning of the etcd instance hosted on the DevMachine's in memory backend.
```
(stole it from apiserver below)